frontend/src/components/History/HistoryDialog.tsx

```diff
@@ -84,15 +85,29 @@ const TryJson = (props: {
   src: string | PostCallResponse;
   collapsed: boolean;
 }) => {
+  const theme = useTheme();
+
   if (typeof props.src === "string") {
     try {
       const parsed = JSON.parse(props.src);
-      return <ReactJson src={parsed} collapsed={props.collapsed} />;
+      return (
+        <ReactJson
+          src={parsed}
+          collapsed={props.collapsed}
+          theme={theme.palette.mode === "dark" ? "summerfruit" : "rjv-default"}
+        />
+      );
     } catch {
       return <code>{props.src}</code>;
     }
   }
-  return <ReactJson src={props.src} collapsed={props.collapsed} />;
+  return (
+    <ReactJson
+      src={props.src}
+      collapsed={props.collapsed}
+      theme={theme.palette.mode === "dark" ? "summerfruit" : "rjv-default"}
```

Comment:
I feel we are repeating this check in every widget? Is this the best approach we can take?  Can it be a js function extracted out?

Reply:
I think i will make a component

Reply:
I have wrapped a `ThemeReactJson`, Now there is no copy + paste.